Neuron/Sections/Transaction/TransactionHistoryPresenter.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  TransactionHistoryPresenter.swift
+//  Neuron
+//
+//  Created by 晨风 on 2018/11/14.
+//  Copyright © 2018 Cryptape. All rights reserved.
+//
+
+import UIKit
+
+protocol TransactionHistoryPresenterDelegate: NSObjectProtocol {
+    func didLoadTransactions(transaction: [TransactionDetails], insertions: [Int], error: Error?)
+    func updateTransactions(transaction: [TransactionDetails], updates: [Int], error: Error?)
+}
+
+class TransactionHistoryPresenter: NSObject, TransactionStatusManagerDelegate {
+    private(set) var transactions = [TransactionDetails]()
+    let token: TokenModel
+    typealias CallbackBlock = ([Int], Error?) -> Void
+    weak var delegate: TransactionHistoryPresenterDelegate?
+    private var hasMoreData = true
+    private var sentTransactions = [TransactionDetails]()
+
+    init(token: TokenModel) {
+        self.token = token
+        walletAddress = WalletRealmTool.getCurrentAppModel().currentWallet!.address
+        tokenAddress = token.address
+        tokenType = token.type
+        super.init()
+        TransactionStatusManager.manager.addDelegate(delegate: self)
+    }
+
+    func reloadData(completion: CallbackBlock? = nil) {
+        guard loading == false else { return }
+        page = 1
+        hasMoreData = true
+        loadMoreData(completion: completion)
+    }
+
+    func loadMoreData(completion: CallbackBlock? = nil) {
+        guard loading == false else { return }
+        guard hasMoreData else { return }
```

Comment:
Please put `else` as separate lines.

```suggestion
        guard hasMoreData else {
           return
        }
```